src/graph/AggregateFunction.h

```diff
@@ -421,6 +423,186 @@ class BitXor final : public AggFun {
     cpp2::ColumnValue     bitXor_;
 };
 
+class CollectList final : public AggFun {
+public:
+    void apply(cpp2::ColumnValue &val) override {
+        if (!has_) {
+            has_ = true;
+            type_ = val.getType();
+            switch(type_) {
+                case ColumnType::int_type:
+                case ColumnType::id_type:
+                case ColumnType::timestamp_type:
+                    type_ = VAR_INT64;
+                    break;
+                case ColumnType::bool_type:
+                    type_ = VAR_BOOL;
+                    break;
+                case ColumnType::float_type:
+                case ColumnType::double_type:
+                    type_ = VAR_DOUBLE;
+                    break;
+                    // TODO suppor string
+//                case ColumnType::str_type:
+//                    type_ = VAR_STR;
+//                    break;
+                default:
+                    return;
+            }
+            cord_ << type_;
+        }
+
+        switch(val.getType()) {
+            case ColumnType::int_type:
+                if (type_ == VAR_INT64) {
+                    cord_ << val.get_integer();
+                }
+                break;
+            case ColumnType::id_type:
+                if (type_ == VAR_INT64) {
+                    cord_ << val.get_id();
+                }
+                break;
+            case ColumnType::timestamp_type:
+                if (type_ == VAR_INT64) {
+                    cord_ << val.get_timestamp();
+                }
+                break;
+            case ColumnType::bool_type:
+                if (type_ == VAR_BOOL) {
+                    cord_ << val.get_bool_val();
+                }
+                break;
+            case ColumnType::float_type:
+                if (type_ == VAR_DOUBLE) {
+                    cord_ << static_cast<double>(val.get_single_precision());
+                }
+                break;
+            case ColumnType::double_type:
+                if (type_ == VAR_DOUBLE) {
+                    cord_ << val.get_double_precision();
+                }
+                break;
+                // TODO suppor string
+//            case ColumnType::str_type:
+//                if (type_ == VAR_STR) {
+//                    cord_ << val.get_str();
+//                }
+//                break;
+            default:
+                return;
+        }
+    }
+
+    cpp2::ColumnValue getResult() override {
+        cpp2::ColumnValue result;
+        result.set_str(cord_.str());
```

Comment:
Why not format the values into string with comma directly?

Reply:
The keywords are provided by users, which are identical to hive sql.